runtime/fs_node.js

```diff
@@ -40,30 +41,31 @@ MlNodeDevice.prototype.exists = function(name) {
   try {
     return this.fs.existsSync(this.nm(name))?1:0;
   } catch (err) {
-    caml_raise_sys_error(err.toString());
+    // TODO: Is it correct to always return a boolean, even on error?
+    return 0;
   }
 }
 MlNodeDevice.prototype.mkdir = function(name, mode) {
   try {
     this.fs.mkdirSync(this.nm(name),{mode:mode});
     return 0
   } catch (err) {
-    caml_raise_sys_error(err.toString());
+    this.raise_unix_exn_of_nodejs_error(err);
```

Comment:
I've another proposal, have mkdir take a third argument and decide what error to raise based on that. What do you think ?

Reply:
I think that will work. My issue was trying to raise the appropriate exception type depending on where this was called from. A flag would work better than the extra try-catch logic I added.